### DISCUSSION TITLE
how to get the length of array？

### DISCUSSION THREAD

**Discussion by eddyheng** [mockoon/mockoon#648] [Help] [answered]
the request body like this:
`{"phoneNumbers": ["15001000012","15100010001","15200010001","15300010001"]}`
The response body is dynamically generated according to the number of phoneNumbers in the request，I write response body like this：
`{
	"returnObjects": [
    {{#repeat (body 'phoneNumbers.length()')}}
    {
      "phonenum": "{{body 'phoneNumbers.{{@index}}'}}", 
      "phone_status": "open"
    }
    {{/ repeat }}
  ]
}`
but it seems that length() cannot get the length of the array
Q1：how do I get the array length?
Q2：if the response 'phone_status' needs to judge according to the requested 'phoneNumbers' and return different values , how should I write the 'phone_status'? like this?
"phone_status": "{{# switch (body 'phoneNumbers.{{@index}}') }}
      {{# case "15001000012" }}"open"{{/ case }}
      {{# case "15100010001" }}"closed"{{/ case }}
      {{# default }}"reviewed"{{/ default }}
    {{/ switch}}"
Q3：Whether regular expressions are supported in #case? like ： {{# case "^1500100" }}"open"{{/ case }}

**Comment by 255kb** [comment, marked as answer]
Q1: 

For the length you need to use the `len` helper with `bodyRaw` as the `body` helper mostly returned stringified data: 
```
{{#repeat (len (bodyRaw 'phoneNumbers'))}}
  {{body (concat 'phoneNumbers.' @index)}}
{{/repeat}}
```
Also, as you can see, `{{ }}` inside `{{ }}`, like in `{{body 'phoneNumbers.{{@index}}'}}` cannot work. You need to use `concat`. 

Q2/Q3: That would be a valid use of a `switch`. Unfortunately, `case` doesn't support regexes. You could try to use [`includes`](https://mockoon.com/docs/latest/templating/mockoon-helpers/#includes). Or you could also randomize the status by using [`oneOf`](https://mockoon.com/docs/latest/templating/mockoon-helpers/#oneof).

[Rules](https://mockoon.com/docs/latest/route-responses/dynamic-rules/) support regexes, so it could solve your issue as you could serve different responses depending on the phone number found. But it's a very different approach.
